src/PageTypes/BasePage.php

```diff
@@ -59,9 +60,13 @@ class BasePage extends SiteTree
         'RelatedPages' => BasePage::class,
     ];
 
+    /**
+     * @var array
+     * @config
+     */
     private static $many_many_extraFields = [
         'RelatedPages' => [
-            'SortOrder' => 'Int',
+            'SortOrder' => DBInt::class,
```

Comment:
This smells more like a bug in the GridFieldSortableRows module than a fix that should be made here, since `Int` is just an injector reference for the DBInt class anyway.

Having said that, a fix in the upstream repo would be backwards compatible with this change anyway, so let's merge it for now!

Reply:
Cheers. OOI where exactly is the automatic Injector switch made when calls for Int are substituted for DBInt for example?

Reply:
I suspect it's in `DataObject::getManyManyComponents` where it returns `Injector::create($someClass)`. The injector alias is defined in `framework/_config/model.yml`